streams/src/test/java/org/apache/kafka/streams/integration/TableTableJoinIntegrationTest.java

```diff
@@ -446,14 +482,18 @@ public void testInnerWithRightVersionedOnly() throws Exception {
                 null,
                 null,
                 null,
-                Collections.singletonList(new TestRecord<>(ANY_UNIQUE_KEY, "D-d", null,  15L)),
-                Collections.singletonList(new TestRecord<>(ANY_UNIQUE_KEY, "E-d", null,  14L)),
                 null,
                 null,
-                Collections.singletonList(new TestRecord<>(ANY_UNIQUE_KEY, "F-d", null,  14L))
+                null,
+                null,
+                Collections.singletonList(new TestRecord<>(ANY_UNIQUE_KEY, "E-e", null,  15L)),
+                Collections.singletonList(new TestRecord<>(ANY_UNIQUE_KEY, null, null,  14L)),
+                null,
+                null,
+                Collections.singletonList(new TestRecord<>(ANY_UNIQUE_KEY, "F-e", null,  14L))
```

Comment:
Should the last two result row flip: we first get `F-e` when we process left hand `F` and get nothing when we process right hand `f`?

Reply:
Yes, you are correct. I've fixed this and also strengthened the check that the test uses (in the latest commit) so that this type of "error" fails the test now. This required updating some of the table-table multi-join expected test results in the process, separate from versioned joins.